File: InClassExamples/oct14/sets.py
# Bob Roos
#
# Demonstration of the "set" data type in Python.
#
# To run this program, type "python sets.py".

a = {1,2,3,4,5,6,7,8,9,10}
b = {i*i for i in range(1,5)}

print "a =",a
print "b =",b
print "a union b =",a | b
print "a intersection b =",a & b
print "a minus b =",a - b
print "b minus a =",b - a
print "elements in a^b:",
for x in a^b:
  print x,
print

File: InClassExamples/oct21/Sets.java
// Bob Roos
// Demonstration of the HashSet class

import java.util.*;

public class Sets {
  public static void main(String[] args) {
    HashSet<Integer> s1, s2;

    s1 = new HashSet<Integer>();
    s2 = new HashSet<Integer>();

    s1.add(1); s1.add(32); s1.add(88); s1.add(-1);
    s2.add(3); s2.add(3); s2.add(3); s2.add(32); s2.add(0);

    for(Integer i : s1) {
       System.out.print(i+" ");
    }
    System.out.println();

    for(Integer i : s2) {
       System.out.print(i+" ");
    }
    System.out.println();
  }
}
